src/views/viewer/ChannelView/ChannelView.tsx

```diff
@@ -155,7 +170,20 @@ export const ChannelView: React.FC = () => {
   }
   const handleOnResizeGrid = (sizes: number[]) => setVideosPerRow(sizes.length)
   const handleChangePage = (page: number) => {
-    isSearching ? setCurrentSearchPage(page) : setCurrentPage(page)
+    if (isSearching) {
+      setCurrentSearchPage(page)
+    } else {
+      setCurrentPage(page)
+      if (!!edges && page * videosPerPage + videosPerPage > edges?.length) {
+        fetchMore({
+          variables: {
+            ...variables,
+            first: page * videosPerPage + videosPerPage * 3,
```

Comment:
What is the reasoning behind that logic? Why not always fetch a constant amount? Currently the number of fetched items will depend on page and can get quite high

Reply:
It's an issue with how `first` variable works, it just starts from the beginning, unless I'm doing something wrong, I have to tell it how many videos I need from page 1 to the page I'm currently at, otherwise, if I go to let's say page 100 and it only loads 50 more videos, so 50 initial + 50 now, it won't be enough videos to populate the page, so that's why I'm not using a constant amount.

Not sure I explained this the best but there it is

Reply:
Also, the `* 3` is there because I load videos for the current page and 2 pages ahead to avoid showing loading components at every page change

Reply:
First shouldn't work this way, you are specifying `after` after all. With this logic you first fetch 50, then 50 + 50, then 50 + 50 + 50. Each fetch should be a separate 50

Reply:
so how does it looks when we start on page 1 then go to page 8 and then go back to page 4? assuming 16 vids per page